benchmark/benchmark_json/src/BM_Cereal_vs_Simdjson_Test.cpp

```diff
@@ -0,0 +1,525 @@
+static void Cereal__Retrive_float_vector(benchmark::State &state)
+{
+    try{
+        // Create a json_string related to test conditions.
+        int i_vec_size = state.range(0);
+        std::string json_string;
+        size_t retrieved_vector_size=0;
+
+        myVectorFloats vec_f(i_vec_size);                                           // Load vector randomly with specified size.
+        std::string json_str = vec_f.serialize__Vector_to__JSON();                  // Now, use this vector to create a JSON serialization memory stream.
+#ifdef SAVE_FLOATS_TO_TEST_FILE                                                     // WARNING: Precompiler variable created through root CMakeLists file.
+        const std::string fileName = {TEST_FILE_PATH_AND_NAME};                     // WARNING: Precompiler variable created through root CMakeLists file.
+        save_to_myfile(json_str, fileName);
+#endif
+        myVectorFloats vec_f_results;                                               // Create an empty vector to load results;
+        for (auto _ : state) 
+        {
+            vec_f_results.deserialize__JSON_to__Vector(json_str);                   // Deserialize JSON memory stream to std::vector<float> with as many floats as in JSON string.
+        }
+        // end timed code.
+        retrieved_vector_size = vec_f_results.get_size();                           // Get number of floats retrieved to results vector.
+        state.counters["retrievedVectorSz"] = retrieved_vector_size;
+
+    }catch(...) {
+        std::cerr << "Controller exception in BM_KPE_166_Retrive_float_vector(): "  << std::endl; 
+    }
+}
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+// benchmarks: Deserialize existing JSON disk file to memory structures.
+//
+// This test benchmarks spent time serializing an existing JSON formatted disk file
+// of std::vector<float>, to a memory std::vector<float> with as many floats as
+// in JSON vector representation loaded.
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- 
+static void Cereal__Retrive_float_vector_from_disk_jsonfile(benchmark::State &state)
+{
+    try{
+        // Create a json_string related to test conditions.
+        int i_vec_size = state.range(0);
+        std::string json_string;
+        size_t retrieved_vector_size=0;
+
+        myVectorFloats vec_f(i_vec_size);                                           // Load vector randomly with specified size.
+        // TODO. Si no existe el archivo, lo creamos.
+        std::string filename = {JSON_FILE_PATH_AND_NAME + std::to_string(i_vec_size)};  // WARNING: Precompiler variable created through root CMakeLists file.
+        vec_f.serialize__Vector_to__JSON_file(filename);                            // Now, use this vector to create a JSON serialization disk file.
+        myVectorFloats vec_f_results;                                               // Create an empty vector to load results;
```

Comment:
rename `vec_f_results` to `deserializedVector`. Please be more careful with the naming of things.